consensus/src/dag/dag_handler.rs

```diff
@@ -47,15 +48,17 @@ impl NetworkHandler {
     async fn start(mut self) {
         while let Some(msg) = self.dag_rpc_rx.next().await {
             if let Err(e) = self.process_rpc(msg).await {
-                warn!(error = ?e, "error sending rpc response for request");
+                warn!(error = ?e, "error processing rpc");
             }
         }
     }
 
     async fn process_rpc(&mut self, rpc_request: IncomingDAGRequest) -> anyhow::Result<()> {
-        let dag_message: DAGMessage = TConsensusMsg::from_network_message(rpc_request.req)?;
+        let dag_message: DAGMessage = rpc_request.req.try_into()?;
 
-        // TODO: verify epoch number and author
+        if dag_message.author()? != rpc_request.sender {
```

Comment:
not every message carries the author field, this basically bails all messages that don't have author field

Reply:
true, here, we expect NodeMsg and CertifiedNodeMsg to have the author? any other message should still be an error?

Reply:
this is the main handler that every message will go through right? if we expect all rpc requests to have an author we should have more explicit error message like "unexpected rpc request"

Reply:
Okay, I made the error explicit now.